lib/src/client.dart

```diff
@@ -463,8 +472,9 @@ class ClientImpl implements Client {
     if (_token != '') {
       request.token = _token;
     }
-    if (_data != null) {
-      request.data = _data!;
+    final data = await _config.getData?.call() ?? _data;
```

Comment:
I think if getData is provided - its returned value should always be used, so it's possible to clean up _data. I.e.:

```
final data = _config.getData != null ? await _config.getData!.call() : _data;
```